src/content/components/SettingsView/SettingsView.js

```diff
@@ -0,0 +1,93 @@
+import React from "react";
+import styles from "./SettingsView.scss";
+import { prefs } from "../../lib/prefs";
+
+export class SettingsView extends React.PureComponent {
+  constructor(props) {
+    super(props);
+    this.state = {
+      bugzilla_email: prefs.get("bugzilla_email"),
+      offline_debug: prefs.get("offline_debug"),
+    };
+    this.onInputChange = this.onInputChange.bind(this);
+    this.onCheckBoxChange = this.onCheckBoxChange.bind(this);
+    this.onPrefChange = this.onPrefChange.bind(this);
+
+    for (const pref of ["bugzilla_email", "offline_debug"]) {
+      prefs.on(pref, this.onPrefChange);
+    }
+  }
+
+  onInputChange(e) {
+    prefs.set(e.target.name, e.target.value);
+  }
+
+  onCheckBoxChange(e) {
+    prefs.set(e.target.name, e.target.checked);
+  }
+
+  onPrefChange({ name, newValue } = {}) {
+    const newState = {};
+    newState[name] = newValue;
+    this.setState(newState);
+  }
+
+  // A button for testing iterations
+  onIterationsCheck() {
+    fetch()
+      .then(rsp => rsp.json())
+      .then(data => console.log(data));
+  }
+
+  componentWillMount() {
+    this.state = {
```

Comment:
Should be `this.setState({...})`

Not your fault, this was already here, but good opportunity to fix it

Reply:
Actually I'm probably the one who copied this lol